### DISCUSSION TITLE
map / combine fields in model

### DISCUSSION THREAD

**Discussion by nidomiro** [prisma/prisma#16765] [Q&A]
Is ist possible to combine fields in prisma?

To get the ts definition:
```ts
type MyType = {
    range: {from: number, to: number}
}
```

out of the model:

```
model MyType {
    rangeFrom: Int
    rangeTo: Int
}
```

**Comment by nurul3101** [comment]
Hello @nidomiro 👋 
Can you please elaborate on your use case?
You can define your own custom type if that's what you meant. Prisma will generate all the types based on your Models, and you can import them like this:
Considering you have a User model:
```typescript
import { PrismaClient, User } from '@prisma/client';
```
<img width="706" alt="Screenshot 2022-12-13 at 2 36 42 PM" src="https://user-images.githubusercontent.com/13049676/207274608-93ad8770-da2d-496f-9386-886ea3b48ff5.png">

**Comment by nidomiro** [comment]
Hi @nurul3101,

thanks for your answer.

The structure I want, but without creating a table for Range is as follows:
```
model User {
    id String    @id @default(uuid())
    saleryRange Range
}

model Range {
    from Int
    to Int
}
```

Currently it is implemented like this to avoid the table generation:
```
model User {
    id String    @id @default(uuid())
    saleryRangeFrom Int
    saleryRangeTo Int
}
```

The model I can use from the generated client should look like:
```ts
type User = {
    id: string
    range: {from: number, to: number}
}
```

maybe https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result will solve this. Does it extend the type `User` when used?

**Reply by nurul3101** [reply]
One alternative I can think of is to define the saleryRange field as JSON column, and define the `from` and `to` fields as keys in the JSON field. This would not require you to create a table, but it would not be type-safe.

Based on your current model, the generated User type would have id, saleryRangeFrom and saleryRangeTo fields. If you need the data in a specific format, you can definitely use [Client Extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result) to create a custom function, and you can create a custom type through which you can validate the return type of this custom function.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
